framework/assets/yii.js

```diff
@@ -143,74 +144,70 @@ window.yii = (function ($) {
                 action = $e.attr('href'),
+                isValidAction = action && action !== '#',
                 params = $e.data('params'),
-                pjax = $e.data('pjax') || 0,
-                usePjax = pjax !== 0 && $.support.pjax,
-                pjaxPushState = !!$e.data('pjax-push-state'),
-                pjaxReplaceState = !!$e.data('pjax-replace-state'),
-                pjaxTimeout = $e.data('pjax-timeout'),
-                pjaxScrollTo = $e.data('pjax-scrollto'),
-                pjaxPushRedirect = $e.data('pjax-push-redirect'),
-                pjaxReplaceRedirect = $e.data('pjax-replace-redirect'),
-                pjaxSkipOuterContainers = $e.data('pjax-skip-outer-containers'),
+                areValidParams = params && $.isPlainObject(params),
+                pjax = $e.data('pjax'),
+                usePjax = pjax !== undefined && pjax !== 0 && $.support.pjax,
                 pjaxContainer,
                 pjaxOptions = {};
 
             if (usePjax) {
-                if ($e.data('pjax-container')) {
-                    pjaxContainer = $e.data('pjax-container');
-                } else {
-                    pjaxContainer = $e.closest('[data-pjax-container=""]');
-                }
-                // default to body if pjax container not found
+                pjaxContainer = $e.data('pjax-container') || $e.closest('[data-pjax-container]');
                 if (!pjaxContainer.length) {
                     pjaxContainer = $('body');
                 }
                 pjaxOptions = {
                     container: pjaxContainer,
-                    push: pjaxPushState,
-                    replace: pjaxReplaceState,
-                    scrollTo: pjaxScrollTo,
-                    pushRedirect: pjaxPushRedirect,
-                    replaceRedirect: pjaxReplaceRedirect,
-                    pjaxSkipOuterContainers: pjaxSkipOuterContainers,
-                    timeout: pjaxTimeout,
+                    push: !!$e.data('pjax-push-state'),
+                    replace: !!$e.data('pjax-replace-state'),
+                    scrollTo: $e.data('pjax-scrollto'),
+                    pushRedirect: $e.data('pjax-push-redirect'),
+                    replaceRedirect: $e.data('pjax-replace-redirect'),
+                    skipOuterContainers: $e.data('pjax-skip-outer-containers'),
+                    timeout: $e.data('pjax-timeout'),
                     originalEvent: event,
                     originalTarget: $e
-                }
+                };
             }
 
             if (method === undefined) {
-                if (action && action != '#') {
-                    if (usePjax) {
-                        $.pjax.click(event, pjaxOptions);
-                    } else {
-                        window.location = action;
-                    }
+                if (isValidAction) {
+                    usePjax ? $.pjax.click(event, pjaxOptions) : window.location.assign(action);
```

Comment:
Use `assign` method for easier stub creation. See http://stackoverflow.com/a/40495796/4323648. It has no difference - https://developer.mozilla.org/ru/docs/Web/API/Location/assign.